api/src/pcapi/scripts/offer/copy_offer_from_venue.py

```diff
@@ -0,0 +1,243 @@
+    for filename in list_filenames:
+        file = os.path.join(EXTRACT_FOLDER, filename)
+        # If file exists, delete it.
+        if os.path.isfile(file):
+            os.remove(file)
+
+
+def _copy_offers(dryrun: bool):
+    description = f"copy offer from venue {FROM_VENUE_ID} to venue {TO_VENUE_ID}"
+    print(description)
+
+    offers_to_copy = (
+        offer_models.Offer.query.join(offer_models.Stock)
+        .filter(offer_models.Offer.venueId == FROM_VENUE_ID)
+        .filter(offer_models.Offer.subcategoryId == subcategories.MATERIEL_ART_CREATIF.id)
+        .filter(offer_models.Offer.validation == offer_mixin.OfferValidationStatus.APPROVED)
+        .filter(sa.or_(offer_models.Stock.remainingQuantity.is_(None), offer_models.Stock.remainingQuantity > 0))
+        .all()
+    )
+
+    print(f"{len(offers_to_copy)} offers to copy")
+
+    _print_offers_csv(offers_to_copy, OFFERS_TO_COPY_BEFORE_CSV)
+
+    if offers_to_copy:
+        stocks_copied = []
+        offers_copied = []
+        all_stocks_to_copy = []
+        i = 0
+        with transaction():
+            for offer_to_copy in offers_to_copy:
+                offer_copied = duplicate_object(offer_to_copy, offer_models.Offer)
+                offer_copied.venueId = TO_VENUE_ID
+                db.session.add(offer_copied)
+
+                offers_copied.append(offer_copied)
+
+                stocks_to_copy: list[offer_models.Stock] = offer_models.Stock.query.filter(
+                    offer_models.Stock.offerId == offer_to_copy.id
+                ).all()
+
+                # print(f"{len(stocks_to_copy)} stocks for offer {offer_to_copy.id}")
+                all_stocks_to_copy.extend(stocks_to_copy)
+
+                price_category_by_offer = {}
+
+                for stock_to_copy in stocks_to_copy:
+                    stock_copied: offer_models.Stock = duplicate_object(stock_to_copy, offer_models.Stock)
+                    stock_copied.quantity = (
+                        None
+                        if stock_to_copy.quantity is None
+                        else stock_to_copy.quantity - stock_to_copy.dnBookedQuantity
+                    )
+                    stock_copied.dnBookedQuantity = 0
+                    stock_copied.offerId = offer_copied.id
+
+                    if stock_to_copy.priceCategoryId:
+                        if stock_to_copy.priceCategoryId not in price_category_by_offer:
+                            priceCategory_to_copy: offer_models.PriceCategory = offer_models.PriceCategory.query.filter(
+                                offer_models.PriceCategory.id == stock_copied.priceCategoryId
+                            ).one()
+                            priceCategory_copied: offer_models.PriceCategory = duplicate_object(
+                                priceCategory_to_copy, offer_models.PriceCategory
+                            )
+                            priceCategory_copied.offerId = offer_copied.id
```

Comment:
Pareil ici, à tester mais je pense plutôt à
```suggestion
                            priceCategory_copied.offer = offer_copied
```